draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -448,12 +456,94 @@ Steps:
 20.  y = u * x (mod p)       // ux
 21.  y = y + v (mod p)       // ux + v
 22. Output (x, y)
-
 ~~~
 
 ### Shallue-Woestijne-Ulas Method {#swu}
 
-((TODO: write this section))
+The Shallue-Woestijne-Ulas (SWU) method, originated in part by
+Shallue and Woestijne {{SW06}} and later simplified and extended by Ulas {{SWU07}},
+deterministically encodes an artbirary string to a point on a curve. 
+This algorithm works for any curve over F_{p^n}. Given curve equation 
+g(x) = x^3 + Ax + B, two separate HashToBase implementations, H0 and H1, 
+this algorithm works as follows:
+
+~~~
+1. t = H0(alpha)
+2. u = H1(alpha)
+3. X1(t, u) = u
+4. X2(t, u) = (-B / A)(1 + 1 / (t^4 * g(u)^2 + t^2 * g(u)))
+5. X3(t, u) = t^3 * g(u)^2  * g(X2(t, u))
```

Comment:
This notation for X1 etc. is a bit confusing. These are effectively inline function definitions, right? I think it might be clearer to just write `X1 = u` and so on.

Reply:
Yeah, they are. I’ll fix it per your suggestion!